src/wallet/rpc/spend.cpp

```diff
@@ -29,16 +29,27 @@ std::vector<CRecipient> ParseRecipients(const UniValue& address_amounts, const s
     std::set<CTxDestination> destinations;
     std::vector<CRecipient> recipients;
     int idx{0};
+    bool has_data{false};
     for (const std::string& address: address_amounts.getKeys()) {
-        CTxDestination dest = DecodeDestination(address);
-        if (!IsValidDestination(dest)) {
-            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, std::string("Invalid Bitcoin address: ") + address);
-        }
-        if (destinations.count(dest)) {
-            throw JSONRPCError(RPC_INVALID_PARAMETER, std::string("Invalid parameter, duplicated address: ") + address);
+        CTxDestination dest;
+        CAmount amount{0};
+        if (address == "data") {
+            if (has_data) {
```

Comment:
`has_data` never set to `true`

Also no tests fail, so it seems there is no test coverage for that

Reply:
Good catch!